tests/unit/lms/services/application_instance_test.py

```diff
@@ -92,44 +92,39 @@ def test_get_by_deployment_id_raises_on_missing(
         with pytest.raises(ApplicationInstanceNotFound):
             service.get_by_deployment_id(issuer, client_id, deployment_id)
 
-    @pytest.mark.parametrize("lms_url", (None, "http://lms-url.com"))
-    @pytest.mark.parametrize("deployment_id", (None, "DEPLOYMENT_ID"))
-    @pytest.mark.parametrize("developer_key", (None, "KEY"))
-    @pytest.mark.parametrize("developer_secret", (None, "SECRET"))
     def test_update_application_instance(
-        self,
-        service,
-        application_instance,
-        lms_url,
-        deployment_id,
-        developer_key,
-        developer_secret,
-        aes_service,
+        self, service, application_instance, aes_service, organization_service
     ):
+        org = factories.Organization()
+        organization_service.get_by_public_id.return_value = org
+
         service.update_application_instance(
             application_instance,
-            lms_url=lms_url,
-            deployment_id=deployment_id,
-            developer_key=developer_key,
-            developer_secret=developer_secret,
+            lms_url="http://example.com",
+            deployment_id="DEPLOYMENT_ID",
+            developer_key="DEVELOPER_KEY",
+            developer_secret="DEVELOPER_SECRET",
+            organization_public_id=mock.sentinel.org_id,
         )
```

Comment:
Test all of this in one go, instead of in a binary combination of every possibility.